### PR TITLE
chore(auth): add fetchauthsession test case generator

### PR DESCRIPTION
**Pull request by sktimalsina**
- [x] PR title and description conform to [Pull Request](https://github.com/aws-amplify/amplify-android/blob/main/CONTRIBUTING.md#pull-request-guidelines) guidelines.

*Issue #, if available:*

*Description of changes:*

*How did you test these changes?*
(Please add a line here how the changes were tested)

- [x] Added Unit Tests
- [ ] Added Integration Tests

*Documentation update required?*
- [ ] No
- [ ] Yes (Please include a PR link for the documentation update)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by banji180 — DISMISSED** [review]

File: aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/featuretest/generators/testcasegenerators/FetchAuthSessionTestCaseGenerator.kt [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+                        ResponseType.Failure,
+                        com.amplifyframework.auth.exceptions.NotAuthorizedException(
+                            cause = notAuthorizedErrorResponse
+                        ).toJsonElement()
+                    )
+                )
+            )
+        }
+
+    // TODO : Fix error case and refresh session case
```
Comment (gpanshu) [inline aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/featuretest/generators/testcasegenerators/FetchAuthSessionTestCaseGenerator.kt:93]: Whats the fix that is required here?

**Review by gpanshu — CHANGES_REQUESTED** [review]
One comment

File: aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/featuretest/generators/testcasegenerators/FetchAuthSessionTestCaseGenerator.kt (hunk 1 above)
Comment (sktimalsina) [reply aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/featuretest/generators/testcasegenerators/FetchAuthSessionTestCaseGenerator.kt:93]: This is broken, as this test cases are not written properly

File: aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/featuretest/generators/authstategenerators/AuthStateJsonGenerator.kt [hunk 2]
```diff
@@ -40,10 +40,16 @@ object AuthStateJsonGenerator : SerializableProvider {
     const val dummyToken = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJ1c2VySWQiLCJ1c2VybmFtZSI6InVzZXJuYW1l" +
         "IiwiZXhwIjoxNTE2MjM5MDIyLCJvcmlnaW5fanRpIjoib3JpZ2luX2p0aSJ9.Xqa-vjJe5wwwsqeRAdHf8kTBn_rYSkDn2lB7xj9Z1xU"
 
+    const val accessKeyId = "someAccessKey"
```
Comment (div5yesh) [inline aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/featuretest/generators/authstategenerators/AuthStateJsonGenerator.kt:43]: Dont we need dummy token in session for test to pass?

Comment (div5yesh) [reply aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/featuretest/generators/authstategenerators/AuthStateJsonGenerator.kt:43]: oh nvm, these are aws credentials.

**Review by div5yesh — DISMISSED** [review]

**Review by gpanshu — DISMISSED** [review]

File: aws-auth-cognito/src/test/java/featureTest/utilities/APICaptorFactory.kt [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+                val consumer = onSuccess[apiName] as Action
+                every { consumer.call() } answers { latch.countDown() }
+                successCaptors[apiName] = actionCaptor
+            }
+            AuthAPI.fetchAuthSession -> {
+                val consumer = onSuccess[apiName] as Action
+                every { consumer.call() } answers { latch.countDown() }
+                successCaptors[apiName] = actionCaptor
+            }
+            else -> throw Error("onSuccess for $authApi is not defined!")
```
Comment (div5yesh) [inline aws-auth-cognito/src/test/java/featureTest/utilities/APICaptorFactory.kt:101]: missing getCurrentUser

**Review by div5yesh — DISMISSED** [review]

**Pull request merged**